packages/chopsticks/src/blockchain/index.ts

```diff
@@ -83,10 +86,21 @@ export class Blockchain {
     if (offchainWorker) {
       this.offchainWorker = new OffchainWorker()
     }
+
+    this.#maxMemoryBlockCount = maxMemoryBlockCount
   }
 
   #registerBlock(block: Block) {
-    this.#blocksByNumber[block.number] = block
+    // refresh key
+    if (this.#blocksByNumber.has(block.number)) {
```

Comment:
why need to delete this first?

Reply:
I was trying to make it like an lru cache. `getBlockAt` and `getBlock` also calls `#registerBlock`, if right now block 10 to 20 are in memory, and user getBlock(10), that means they need to access it, so I re-insert it into the `Map`, so the first one to be deleted will be block 11, and block 10 is still in memory for faster access

If we always want to delete the smallest block number, then we don't need this.

Reply:
wait, I was wrong, `#registerBlock` is only called when block is not found, so we actually don't need this😅